File: 20-Apr-Java-7.30-Mihir/Classwork/LectureFlow/ConstructorDemo.java

class Box{
	double width, height, depth;
	Box(){
		System.out.println("Default Constructor Called");
		width = 5;
		height = 4;
		depth = 3;
	}
	
	Box(double w, double h, double d){
		System.out.println("Parameterized Constructor Called");
		width = w;
		height = h;
		depth = d;		
	}
	
	Box(Box b){
		System.out.println("Copy Constructor Called");
		width = b.width;
		height = b.height;
		depth = b.depth;
	}
	
	void volume() {
		System.out.println("Volume is "+(width*height*depth));		
	}
	
	
}



public class ConstructorDemo{
	public static void main(String[] args) {
	Box b1 = new Box();
	b1.volume();
	Box b2 = new Box(7,6,3);
	b2.volume();
	Box b3 = new Box(b1);
	b3.volume();
	
	}
	
}

File: 28-Apr-C-C++-DB/Classwork/C++_Lecture_Flow/11-6-22/constructordemo.cpp
/*
Constructor : 1) Constructor same name as Class name;
			  2)No Return-type;
			  3)It is automatically Called/Invoke when object of class 
			    is created.
			 4) It is used to intialize values;
			 
	Types of Constructor : 1) Default Constructor
					      2) Parameterized Constructor  :   Value as Parameter;
					      3)Copy Constructor		    :   Object as Parameter;				
			
*/

#include<iostream>
using namespace std;

class Box{
	public:
		double width ,height,depth;
		Box()
		{
			cout<<"\n\nDefault Constructor Called."<<endl;
			width = 10;
			height = 7;
			depth = 5;
		}
		
		Box(double w,double h,double d)
		{
			cout<<"\n\nParameterized Constructor Called."<<endl;
			width = w;
			height = h;
			depth = d;
		}
		
		Box(const Box &tr)
		{
			cout<<"\n\nCopy Constructor Called."<<endl;
			width = tr.width;
			height = tr.height;
			depth = tr.depth;
		}
		
	 void volume()
		{
			cout<<"Volume is "<<(width*height*depth);
		}
};

int main()
{
	Box a1;         //Default Constructor Called.
	a1.volume();
	
	Box a2(10,32,4);   //Parameterized Constructor Called.
	a2.volume();

	Box b3(a1);			//Copy Constructor Called.
	b3.volume();
	
	return 0;
}
